dotnetv4/CloudWatchLogs/LargeQuery/Scenarios/LargeQueryWorkflow.cs

```diff
@@ -0,0 +1,838 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// snippet-start:[CloudWatchLogs.dotnetv4.LargeQueryWorkflow]
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+using Amazon.CloudFormation;
+using Amazon.CloudFormation.Model;
+using Amazon.CloudWatchLogs;
+using Amazon.CloudWatchLogs.Model;
+using CloudWatchLogsActions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace CloudWatchLogsScenario;
+
+public class LargeQueryWorkflow
+{
+    /*
+    Before running this .NET code example, set up your development environment, including your credentials.
+    This .NET code example performs the following tasks for the CloudWatch Logs Large Query workflow:
+
+    1. Prepare the Application:
+       - Prompt the user to deploy CloudFormation stack and generate sample logs.
+       - Deploy the CloudFormation template for resource creation.
+       - Generate 50,000 sample log entries using CloudWatch Logs API.
+       - Wait 5 minutes for logs to be fully ingested.
+
+    2. Execute Large Query:
+       - Perform recursive queries to retrieve all logs using binary search.
+       - Display progress for each query executed.
+       - Show total execution time and logs found.
+
+    3. Clean up:
+       - Prompt the user to delete the CloudFormation stack and all resources.
+       - Destroy the CloudFormation stack and wait until removed.
+    */
+
+    public static ILogger<LargeQueryWorkflow> _logger = null!;
+    public static CloudWatchLogsWrapper _wrapper = null!;
+    public static IAmazonCloudFormation _amazonCloudFormation = null!;
+
+    private static string _logGroupName = "/workflows/cloudwatch-logs/large-query";
+    private static string _logStreamName = "stream1";
+    private static long _queryStartDate;
+    private static long _queryEndDate;
+
+    public static bool _interactive = true;
+    public static string _stackName = "CloudWatchLargeQueryStack";
+    private static string _stackResourcePath = "../../../../../../../scenarios/features/cloudwatch_logs_large_query/resources/stack.yaml";
```

Comment:
Because the path is relative, it didn't work for me. On changing the path, the entire scenario worked for me. Can it be configured so that it works for anyone?

Reply:
Also, is it possible that going ahead, the scenarios directory contains only specs? The entire scenarios directory gets synced with its knowledge base and adding yaml or other file types can hinder syncs.